src/kagglehub/clients.py

```diff
@@ -143,6 +158,11 @@ def download_file(self, path: str, out_file: str, resource_handle: Optional[Reso
             total_size = int(response.headers["Content-Length"])
             size_read = 0
 
+            if isinstance(resource_handle, CompetitionHandle) and not _download_needed(
```

Comment:
Why is this something we only do for competitions?  It seems like returning early would be good for all downloads, right?